### DISCUSSION TITLE
My LG C2 does not wake up from sleep, but it does go to sleep  (extra network card, broadcast on wrong interface problem ?)

### DISCUSSION THREAD

**Discussion by shodanx2** [JPersson77/LGTVCompanion#177] [General]
Hi,

I think the issue is that LGTVCompanion is not sending the WOL packets to the correct interface. 

I have one onboard ethernet connected to my lan and with internet access.

I have one virtual network card for virtualbox 

And I have a usb to network card connected to only the TV. (I don't want the TV to have internet access at all)

When I did the pairing, the TV did ask to accept the pairing.
But now that it is done, I have hard time telling, does LGTVC still talk to the TV ?

It is not clear in the interface whether this is still connected ? Except when I run my "LCDOff.exe" program, the TV immediately goes to sleep and the red light turns off. But there's no other clear signs.

**Comment by JPersson77** [comment]
Hi, the app establish a new connection as needed (i.e it does  not remain connected) 

It sounds like the app can vommunicate with the TV. Can you confirm the TV / usb-network card is configured to wake on  lan?  

Please enable logging if you haven't already. It can be useful for understanding an issue

**Reply by shodanx2** [reply]
Hello,
I believe the Wake-on-lan packets are coming out the wrong interface.
I will setup a way to sniff the network on the side of the "TV" network to confirm

(and I am having similar issues with gstreamer sending multicast to the wrong interface too, seems broadcast and multicast in windows are router by something called "metric")

Is there a way to tell LGTVC to force send the WOL packets out of a specific interface ?

**Reply by GorgiGR** [reply]
> Hello, I believe the Wake-on-lan packets are coming out the wrong interface. I will setup a way to sniff the network on the side of the "TV" network to confirm
> 
> (and I am having similar issues with gstreamer sending multicast to the wrong interface too, seems broadcast and multicast in windows are router by something called "metric")
> 
> Is there a way to tell LGTVC to force send the WOL packets out of a specific interface ?

Hey, did you come any further on this issue?

My LG G4 has exactly the same issue. When testing it shuts down just fine, however I did not manage to wake it up (whether with automatic or Network-wide or Local Broadcast packets or unicast packet).
I have two ethernet cards also, so I would be really interested in how you solved your problem, if you did.

**Comment by JPersson77** [comment]
@hi @GorgiGR can you try to clear the ARP cache and/or manually remove entries related to the IP/MAC of the TV to see if that changes anything?

in an ADMIN CMD you can use the following commands:

"arp-a" will list all entries in the ARP cache
"netsh interface ip delete arpcache" will clear the netcache, but may (depending on) not clear everything
"arp -d XXX.XXX.XXX.XXX" will delete entry with inet_addr (IP) XXX.XXX.XXX.XXX

Edit: the LGTVC debug-level log will show the selected interface, as returned by getBestRoute2(), in a log message like this:
`ARP override best source: 192.168.1.150, Interface index: 6, LUID: 1689399632855040, Route protocol: 2`
This in used to create a transient ARP entry in the ARP cache. Maybe something has gone wrong and there is an old entry

**Reply by GorgiGR** [reply]
For anyone reading this, my problem was after all of a different nature.
I have a VPN client running which supports split tunneling, and I already had configured both the UI and the service components of LGTVCompanion to NOT be routed through the VPN tunnel, which was the reason why I thought this wasn't the problem. Quite by chance I found out that when not using the VPN client the TV wake up call works! 
After taking a look at the sources, I found out that the wake-up call is executed in the CLI ( **Edited: this is not the case after all!!! See comments below!**) component of the LGTVCompanion, so the solution for me was to include also this executable into the group of apps using split tunelling in my VPN settings. Now everything works as expected.

@JPersson77 great job man, this handy piece of software might actually double my G4's life :) Thanks for creating this gem!

**Comment by JPersson77** [comment]
Hi @GorgiGR thanks for the feedback and the kind words! The WOL is however sent out by the service (LGTVsvc.exe) and the functionality does not depend on LGTVcli.exe. But I don't know how it started working for you when you added the cli to the whitelist 

@shodanx2 Incidentally, I have added the functionality to manually choose which network adapter to send the WOL from to the code. Going to publish a development build on discord later today https://discord.gg/7KkTPrP3fq if you want to try it

**Reply by GorgiGR** [reply]
@JPersson77 I just checked and yes, it seems to be working now, even if the CLI is not included in the split tunelling app list 🤷‍♂️
This is super weird, I cannot explain what happened yesterday. The reason why I thought it was the CLI was that somewhere around line 1300 there is a "spawn the WOL thread" entry in the CLI sources.

The take away from this story is "If you have a VPN just disable it for every component of LGTVCompanion to be sure" 😁

**Reply by JPersson77** [reply]
Agreed :)

**Comment by JPersson77** [comment]
Version 5 was just released which allows for selecting the network adapter to send WOL packets from. Let me know how it works for you please

**Reply by GorgiGR** [reply]
Just tested with v5.0.2 (The new option detecting video running in browser addresses one of my main headaches btw 😁 )
Unfortunately, even in this version and even if I explicitly select the Ethernet interface for the wake up packets the G4 is not waking up as it should if I don't use split tunneling for LGTVCompanion executables. I tested all the options (Broadcast. Local Broadcast, Unicast).
To all people using VPNs, **include the LGTVCompanion executables in your VPN whitelist** 

@JPersson77 maybe it would be helpful to mention this in the readme?
